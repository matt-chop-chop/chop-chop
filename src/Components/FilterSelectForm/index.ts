export { default as FilterSelectForm } from "./FilterSelectForm";
