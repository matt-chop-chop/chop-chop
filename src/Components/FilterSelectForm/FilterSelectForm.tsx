import { FormControl, FormErrorMessage, FormLabel } from "@chakra-ui/react";
import { FilterSelect } from "@/Components";
import React from "react";

export type FilterSelectFormProps = {
  label: string;
  id: string;
  loading: boolean;
  options: (Ingredient | Category | Area)[];
  setSelection: (selection: string) => void;
  errorMessage: string;
  error?: Error;
};

const FilterSelectForm = ({
  label,
  id,
  loading,
  options,
  setSelection,
  errorMessage,
  error,
}: FilterSelectFormProps) => {
  return (
    <FormControl isInvalid={!!error} mt={5}>
      <FormLabel htmlFor={id} variant="body">
        {label}
      </FormLabel>
      <FilterSelect
        id={id}
        isDisabled={!!error}
        isLoading={loading}
        options={options}
        onSelectedChange={setSelection}
      />
      <FormErrorMessage>
        {errorMessage}
        {error?.message && `: ${error.message}`}
      </FormErrorMessage>
    </FormControl>
  );
};

export default FilterSelectForm;
