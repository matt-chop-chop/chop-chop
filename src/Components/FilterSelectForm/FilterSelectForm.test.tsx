import { shallow } from "enzyme";
import React from "react";
import FilterSelectForm, { FilterSelectFormProps } from "./FilterSelectForm";

const setSelection = jest.fn();

const props: FilterSelectFormProps = {
  label: "This is a test",
  id: "test",
  loading: false,
  options: [{ name: "Test 1" }, { name: "Test 2" }],
  setSelection: setSelection,
  errorMessage: "Test Error",
  error: new Error("There has been an error"),
};

describe("FilterSelectForm", () => {
  it("is exported from FilterSelectForm", () => {
    expect(FilterSelectForm).toBeDefined();
  });

  it("renders as expected ", () => {
    const wrapper = shallow(<FilterSelectForm {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("calls setSelection when a selection is made", () => {
    const wrapper = shallow(<FilterSelectForm {...props} />);
    const FilterSelect = wrapper.find("FilterSelect");
    expect(FilterSelect).toHaveLength(1);
    const onSelectedChange = FilterSelect.prop("onSelectedChange") as (
      selection: string
    ) => void;
    onSelectedChange("test");
    expect(setSelection).toHaveBeenCalledWith("test");
  });
});
