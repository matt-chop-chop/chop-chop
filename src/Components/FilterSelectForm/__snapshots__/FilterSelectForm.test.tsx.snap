// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterSelectForm renders as expected  1`] = `
<FormControl
  isInvalid={true}
  mt={5}
>
  <FormLabel
    htmlFor="test"
    variant="body"
  >
    This is a test
  </FormLabel>
  <FilterSelect
    id="test"
    isDisabled={true}
    isLoading={false}
    onSelectedChange={[MockFunction]}
    options={
      [
        {
          "name": "Test 1",
        },
        {
          "name": "Test 2",
        },
      ]
    }
  />
  <FormErrorMessage>
    Test Error
    : There has been an error
  </FormErrorMessage>
</FormControl>
`;
